### PR TITLE
NMS-13540: Highlite search terms

### PR DESCRIPTION
**Pull request by fooker**
JIRA: https://issues.opennms.org/browse/NMS-13540

### CODE REVIEW

File: supplemental-ui/js/vendor/search.js [hunk 1]
```diff
@@ -193,17 +197,153 @@ window.antoraLunr = (function (lunr) {
...
+                  }
+
+                  // Reset blank status
+                  blank = false
+
+                  // Check if match is part of this node
+                  if (match[0] < index + text.length) {
+
+                    // Define range of match in relation to current node
+                    let range = document.createRange()
```
Comment (RangerRick) [inline supplemental-ui/js/vendor/search.js:310]: neat, I did not know about ranges before

**Review by RangerRick — APPROVED** [review]
nothing wrong jumps out at me about this.  Can you describe how this solves the issues you ran into with the index/offsets not matching up with the rendered HTML?

**Comment by fooker** [comment]
> nothing wrong jumps out at me about this. Can you describe how this solves the issues you ran into with the index/offsets not matching up with the rendered HTML?

@RangerRick My first assumption was wrong. The index is build using the rendered HTML To do so, the indexer takes the DOM, removes the titles (and some other stuff) and uses jquery.text to extract the HTML. Afterwards it strips the tags and removes leading white-spaces.

The code resembles the HTML processing while walking the DOM. Main part is to reduce multiple nodes containing leading white-spaces to a single white-space while skipping certain tags for titles.

**Comment by RangerRick** [comment]
Aha! Nice, so once you duplicate the "removed" tokens it matches. Good catch.

**Comment by Bonrob2** [comment]
@fooker since Ronny is away, can we merge this PR, or do we need a second reviewer besides Ben?

**Comment by fooker** [comment]
> @fooker since Ronny is away, can we merge this PR, or do we need a second reviewer besides Ben?

I think so.

**Pull request merged**